src/RecursiveIteratorAggregateIterator.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare(strict_types=1);
+
+namespace loophp\iterators;
+
+use Generator;
+use Iterator;
+use IteratorAggregate;
+use RuntimeException;
+use Traversable;
+
+use function count;
+
+/**
+ * @template TKey
+ * @template T
+ *
+ * @implements IteratorAggregate<int, T>
+ */
+class RecursiveIteratorAggregateIterator implements IteratorAggregate
+{
+    /**
+     * @var list<Iterator>
+     */
+    private ?array $stack = null;
+
+    /**
+     * @param Traversable<TKey,T> $input
+     */
+    public function __construct(private readonly Traversable $input) {}
+
+    public function getDepth(): int
+    {
+        return null === $this->stack ? -1 : count($this->stack);
+    }
+
+    /**
+     * @return Generator<int,T>
+     */
+    public function getIterator(): Generator
+    {
+        if (null !== $this->stack) {
+            throw new RuntimeException('Iterator already active ');
```

Comment:
This cannot happen. This method always returns a new iterator, ready to be consumed.

Reply:
```
foreach($it as $a){ //generator assign stack to iterator
    foreach($it as $b){ //same iterator, new stack will replace original, throw exception
    }
}
```

Reply:
What are you trying to tell me here?

Reply:
this Exception will be thrown in this case